jazz_deployments-event-handler/index.js

```diff
@@ -0,0 +1,383 @@
+		}
+		Promise.all(processEventRecordPromises)
+			.then((result) => {
+				return resolve(result);
+			})
+			.catch((error) => {
+				logger.error("processEventRecords failed" + JSON.stringify(error));
+				return reject(error);
+			});
+	});
+};
+
+var processEventRecord = (record, configData, authToken) => {
+	return new Promise((resolve, reject) => {
+		var sequenceNumber = record.kinesis.sequenceNumber;
+		var encodedPayload = record.kinesis.data;
+		var payload;
+		return checkForInterestedEvents(encodedPayload, sequenceNumber, configData)
+			.then(result => {
+				payload = result.payload;
+				if (result.interested_event) {
+					return processEvent(payload, configData, authToken);
+				} else {
+					return new Promise((resolve, reject) => {
+						resolve({ "message": "Not an interesting event" });
+					});
+				}
+			})
+			.then(result => {
+				handleProcessedEvents(sequenceNumber, payload);
+				return resolve(result);
+			})
+			.catch(err => {
+				logger.error("processEventRecord failed for " + JSON.stringify(record));
+				handleFailedEvents(sequenceNumber, err.failure_message, payload, err.failure_code);
+				return reject(err);
+			});
+	});
+};
+
+var checkForInterestedEvents = (encodedPayload, sequenceNumber, configData) => {
+	return new Promise((resolve, reject) => {
+		var kinesisPayload = JSON.parse(new Buffer(encodedPayload, 'base64').toString('ascii'));
+		if (kinesisPayload.Item.EVENT_TYPE && kinesisPayload.Item.EVENT_TYPE.S) {
+			if (_.includes(configData.EVENTS.EVENT_TYPE, kinesisPayload.Item.EVENT_TYPE.S) &&
+				_.includes(configData.EVENTS.EVENT_NAME, kinesisPayload.Item.EVENT_NAME.S)) {
+				logger.info("found " + kinesisPayload.Item.EVENT_TYPE.S + " event with sequence number: " + sequenceNumber);
+				return resolve({
+					"interested_event": true,
+					"payload": kinesisPayload.Item
+				});
+			} else {
+				logger.error("Not an interested event or event type");
+				return resolve({
+					"interested_event": false,
+					"payload": kinesisPayload.Item
+				});
+			}
+		}
+	});
+};
+
+var processEvent = (eventPayload, configData, authToken) => {
+	return new Promise((resolve, reject) => {
+		if (eventPayload.SERVICE_CONTEXT && eventPayload.SERVICE_CONTEXT.S) {
+			if (eventPayload.EVENT_NAME.S === configData.EVENTS.create_event_name) {
+				processCreateEvent(eventPayload, configData, authToken)
+					.then(result => { return resolve(result); })
+					.catch(err => {
+						logger.error("processCreateEvent failed: " + err);
+						return reject(err);
+					});
+			} else if (eventPayload.EVENT_NAME.S === configData.EVENTS.update_event_name) {
+				processUpdateEvent(eventPayload, configData, authToken)
+					.then(result => { return resolve(result); })
+					.catch(err => {
+						logger.error("processUpdateEvent failed: " + err);
+						return reject(err);
+					});
+			} else if (eventPayload.EVENT_NAME.S === configData.EVENTS.delete_event_name) {
+				//TODO
```

Comment:
can you put a task/story number here if this has to be done or remove this code

Reply:
This part not handled yet. So as Surya suggested I put TODO

Reply:
sure. my point was to add a task/story in rally and then put that number here(US...). this way we will know what the todo is and it wont be lost and can be easily tracked.